subprojects/testing-base/src/main/java/org/gradle/api/tasks/testing/AbstractTestTask.java

```diff
@@ -494,6 +527,20 @@ private void createReporting(Map<String, TestClassResult> results, TestOutputSto
         }
     }
 
+    /**
+     * Sets the test name patterns to be included in execution.
+     * Classes or method names are supported, wildcard '*' is supported.
+     * For more information see the user guide chapter on testing.
+     *
+     * For more information on supported patterns see {@link TestFilter}
+     */
+    @Option(option = "tests", description = "Sets test class or method name to be included, '*' is supported.")
+    @Incubating
+    public T setTestNameIncludePatterns(List<String> testNamePattern) {
+        filter.setCommandLineIncludePatterns(testNamePattern);
+        return taskType.cast(this);
```

Comment:
Should we just change this method so it's a pure setter (no return value) since it's incubating? It would remove all the templating that we have to do.

Reply:
I don't think we need to do this with a generic parameter.  You can declare this as:

`public AbstractTestTask setTestNameIncludePatterns(List<String> testNamePattern)`

And use overrides in the subclasses to return the more specific type.

Reply:
Good idea, done.